core/controllers/diagnostic_test_player.py

```diff
@@ -77,30 +104,52 @@ def get(self, topic_id: str) -> None:
 
         diagnostic_test_skill_ids = topic.skill_ids_for_diagnostic_test
 
-        # From each skill, two questions were fetched. The first question
-        # (main question) will be presented to the learner initially and if the
-        # learner attempted incorrectly, then another question from the same
-        # skill (backup question) will be presented otherwise not. Hence the
-        # number of questions is twice the length of skill IDs.
-        question_count = len(diagnostic_test_skill_ids) * 2
-
+        skill_id_to_all_questions_fetchable_at_one_time = (
```

Comment:
I don't understand this concept of "fetchable at one time". What does it mean? Could you perhaps try explaining it in a different way?

Reply:
Removed this logic and updated the implementation as per the discussion on [this](https://github.com/oppia/oppia/pull/16408#discussion_r1014605155) comment.